### PR TITLE
Adding callback instead of throw an error.

### PR DESCRIPTION
**Pull request by alansouzati**
Fixes https://github.com/yuanqing/gulp-tape/issues/19

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -60,10 +60,11 @@ var gulpTape = function(opts) {
         this.fail = 0;
         this.pass = 0;
         tapeStream.push(null);
-        callback();
 
         if (shouldErrorOut) {
-          throw new Error('Test failed');
+          callback(new PluginError(PLUGIN_NAME, 'Test failed'));
+        } else {
+          callback();
```
Comment (yuanqing) [inline index.js]:
``` js
callback(shouldErrorOut ? new PluginError(PLUGIN_NAME, 'Test failed') : null);
```

**Comment by yuanqing** [comment]
@alansouzati 
1. Thank you, and apologies. Should probably have done a proper PR for my change (else suggested the change in your branch before merging)
2. Please update `test/gulpfile.js` to add this bit to the `test` task:
   
   ``` js
   .on('error', function(error) {
     gutil.log(error.message);
     process.exit(1);
   });
   ```

**Comment by alansouzati** [comment]
I added a separate test for bail, but I'm not sure if this is how we should test it. But for now I think we can live with it. 

In a near future I will try and add tests for your library ;)

**Comment by alansouzati** [comment]
hi @yuanqing do you want me to go ahead and merge this PR?

File: test/fixtures/test-bail/index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+var test = require('tape');
+var abs = require('../');
```
Comment (yuanqing) [inline test/fixtures/test-bail/index.js]: Remove because it's unused

File: test/gulpfile.js [hunk 3]
```diff
@@ -6,10 +6,20 @@ var tapColorize = require('tap-colorize');
 
 var tape = require('../');
 
+gulp.task('test-bail', function() {
+  return gulp.src('fixtures/test-bail/*.js')
+    .pipe(tape({
+      bail: true,
+      reporter: tapColorize()
+    })).on('error', function(error) {
+      console.log('callback invoked');
```
Comment (yuanqing) [inline test/gulpfile.js]:
``` js
gutil.log(error.message);
```

Add `var gutil = require('gulp-util');` to the top of the file

File: test/fixtures/test-bail/index.js (hunk 2 above)
Comment (alansouzati) [inline test/fixtures/test-bail/index.js]: true sorry for missing that

**Comment by yuanqing** [comment]
In order for the build to pass, we need to add a `|| true` at the end of the `test` script:

``` json
"test": "gulp --cwd test test && gulp --cwd test istanbul && gulp --cwd test test-bail || true"
```

Adding a `|| true` seems to defeats the purpose of having the `test-bail` test actually. Alternative is to remove the `gulp --cwd test test-bail` bit

**Comment by alansouzati** [comment]
I agree, let me remove the test and work in a separate PR to refactor the test structure.

**Comment by alansouzati** [comment]
hey @yuanqing i think it should be good to go for now.

**Pull request merged**